crates/turborepo-lockfiles/src/berry/mod.rs

```diff
@@ -1172,4 +1222,480 @@ mod test {
         let lockfile = BerryLockfile::new(data, None).unwrap();
         assert_eq!(lockfile.turbo_version().as_deref(), Some("1.4.6"));
     }
+
+    #[test]
+    fn test_resolution_pruning_preserves_both_original_and_resolved_descriptors() {
+        // This test verifies the fix for GitHub issue #2791:
+        // "pruned lockfile missing resolution entries"
+        // When yarn resolutions override dependencies, both the original descriptor
+        // (as used by packages) and the resolved descriptor should be preserved
+
+        // Use the existing robust test data and add our own resolution
+        let data = LockfileData::from_bytes(include_bytes!(
+            "../../fixtures/robust-berry-resolutions.lock"
+        ))
+        .unwrap();
+
+        // Create resolutions that override ajv to use version 8 instead of 6
+        let manifest = BerryManifest {
+            resolutions: Some(
+                [("ajv".to_string(), "^8".to_string())]
```

Comment:
We need to actually recreate the fixture lockfile and not just set `resolutions` here.